Update RELEASE.md sync doc (#768)
conda/infrastructure: .github/sync/RELEASE.md

<!-- These docs are updated and synced from https://github.com/conda/infra -->

<!-- links -->
<!-- (some of these have been commented out for now, until the `repo.url` syntax works) -->
<!-- [epic template]: {{ repo.url }}/issues/new?assignees=&labels=epic&template=epic.yml
[compare]: {{ repo.url }}/compare

> [new release]: {{ repo.url }}/releases/new -->

Reviewer: instead of commenting this out can we just put the full url here and avoid making all of the url changes inline?

[infrastructure]: https://github.com/conda/infrastructure
[rever docs]: https://regro.github.io/rever-docs
[release docs]: https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

# Release Process

> **Note**
> Throughout this document are references to the version number as `YY.M.0`, this should be replaced with the correct version number. Do **not** prefix the version with a lowercase `v`.

## 1. Open the release issue and cut a release branch. (do this ~1 week prior to release)

> **Note**

> > The [epic template](https://github.com/conda/infrastructure/issues/new?assignees=&labels=epic&template=epic.yml) is perfect for this, just remember to remove the **`epic`** label.

Reviewer 2: I'd go with either two separate sentences or turn that comma into a semicolon.

    <!-- link any other feedstock PRs here -->
- [ ] Hand off to the Anaconda packaging team
- [ ] Announce release
    - Blog Post (optional)
        - [ ] conda.org (link to pull request)
    - Long form
        - [ ] Create release [announcement draft](https://github.com/conda/communications)
        - [ ] [Discourse](https://conda.discourse.group/)

>         - [ ] Matrix

Reviewer: ```suggestion
        - [ ] [Matrix (conda/conda)](https://matrix.to/#/#conda_conda:gitter.im) (this auto posts from Discourse)
```

    - Summary

>         - [ ] Twitter

Reviewer: ```suggestion
        - [ ] [Twitter](https://twitter.com/condaproject)
```

```
</details>

> **Note**

> > The new release branch should adhere to the naming convention of `YY.M.x`

Reviewer 2: ```suggestion
> The new release branch should adhere to the naming convention of `YY.M.x`.
```

Totally understand if we feel leaving the period out is less confusing for formatting reasons. I forget, do the single backticks do italics or code formatting? If the latter, the period won't be a problem, IMO.

> ## 2. Alert various parties of the new release. (do this ~1 week prior to release)

Reviewer: ```suggestion
## 2. Alert various parties of the upcoming release. (do this ~1 week prior to release)
```

> Let various interested parties know about the new release; at minimum, conda-forge maintainers should be informed about new releases. For major releases, a blog post describing the new features should be created and posted.

Reviewer: ```suggestion
Let various interested parties know about the upcoming release; at minimum, conda-forge maintainers should be informed. For major features, a blog post describing the new features should be prepared and posted once the release is completed (see the announcements section of the release issue).
```

> 10. Update Release Issue to include a link to the Release PR.

Reviewer 2: I'm seeing some capitalization inconsistencies with Release Issue and Release PR. Not a critical thing, but thought I'd point it out.

Reviewer 2: Release branch appears to be all lowercase throughout. Any reason to your knowledge that these are capitalized?

> 10. Update Release Issue to include a link to the Release PR.

Author: ```suggestion
10. Update release issue to include a link to the release PR.
```

11. [Create](https://github.com/conda/infrastructure/releases/new) the release and **SAVE AS A DRAFT** with the following values:

    > **Note**

>     > Only publish the release after the Release PR is merged, until then always **save as draft**.

Author: ```suggestion
    > Only publish the release after the release PR is merged, until then always **save as draft**.
```

</details>

> ## 5. Wait for review and approval of Release PR.

Author: ```suggestion
## 5. Wait for review and approval of release PR.
```

## 6. Merge release PR and publish release.

## 7. Merge/cherry pick the release branch over to the `main` branch.

<details>
<summary>Internal process</summary>

1. From the main "< > Code" page of the repository, select the drop down menu next to the `main` branch button and then select "View all branches" at the very bottom.

2. Find the applicable `YY.MM.x` branch and click the "New pull request" button.

3. "Base" should point to `main` while "Compare" should point to `YY.MM.x`.

4. Ensure that all of the commits being pulled in look accurate, then select "Create pull request".

> **Note**
> Make sure NOT to push the "Update Branch" button. If there are [merge conflicts](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/about-merge-conflicts), create a temporary "connector branch" dedicated to fixing merge conflicts separately from the `YY.M.0` and `main` branches.

5. Review and merge the pull request the same as any code change pull request.

> **Note**

> > The commits from the release branch need to be retained in order to be able to compare individual commits; in other words, a "merge commit" is required when merging the resulting pull request vs. a "squash merge". Protected branches will require permissions to be temporarily updated in order to enable this action.

Reviewer: ```suggestion
> The commits from the release branch need to be retained in order to be able to compare individual commits; in other words, a "merge commit" is required when merging the resulting pull request vs. a "squash merge". Protected branches will require permissions to be temporarily relaxed in order to enable this action.
```

</details>

## 10. Continue championing and shepherding.

> Remember to make all relevant announcements and continue to update the Release Issue with the latest details as tasks are completed.

Author: ```suggestion
Remember to make all relevant announcements and continue to update the release Issue with the latest details as tasks are completed.
```

## 10. Continue championing and shepherding.

> Remember to make all relevant announcements and continue to update the release Issue with the latest details as tasks are completed.

Reviewer 2: ```suggestion
Remember to make all relevant announcements and continue to update the release issue with the latest details as tasks are completed.
```
